### PR TITLE
Added: Support for Create app and get all apps via api

### PR DESCRIPTION
**Pull request by mahavirvataliya**
Added: get all apps via secure header

### CODE REVIEW

**Comment by dimiro1** [comment]
@mahavirvataliya Thanks for your contribution. I will have a look at your PR.

File: go.mod [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module ipe
 
+go 1.14
```
Comment (dimiro1) [inline go.mod]: Can you change it to 1.11?

File: ipe.go [hunk 2]
```diff
@@ -75,6 +77,39 @@ func Start(filename string) {
...
 	)
 
+	router.Path("/app/create").Methods("POST").Handler(
+		api.NewCreateApplication(inMemoryStorage),
+	)
+
+	router.HandleFunc("/apps/all",
+		func(w http.ResponseWriter, r *http.Request) {
+			authHeader := r.Header.Get("Authorization")
+			log.Error(os.Getenv("AUTH_TOKEN"))
```
Comment (dimiro1) [inline ipe.go]: Please remove this log line.

File: README.md [hunk 3]
```diff
@@ -142,6 +144,113 @@ let pusher = new Pusher({
...
+
+```
+curl --location --request GET 'http://127.0.0.1:8080/apps/all' \
+--header 'Authorization: 7kAWg6ADHx'
+```
+
+Respose will be
+
+```
+{
```
Comment (dimiro1) [inline README.md:223]: This JSON response is invalid.

File: ipe.go [hunk 4]
```diff
@@ -75,6 +77,39 @@ func Start(filename string) {
...
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusUnauthorized)
+				w.Write([]byte(`{ "Error":  "Unauthorized access" }`))
+				return
+			}
+
+			out, err := json.Marshal(inMemoryStorage.GetAllApps())
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte(`{ ` + string(out) + ` }`))
```
Comment (dimiro1) [inline ipe.go:97]: This will return invalid JSON.

File: ipe.go [hunk 5]
```diff
@@ -75,6 +77,39 @@ func Start(filename string) {
 		websockets.NewWebsocket(inMemoryStorage),
 	)
 
+	router.Path("/app/create").Methods("POST").Handler(
+		api.NewCreateApplication(inMemoryStorage),
```
Comment (dimiro1) [inline ipe.go:81]: Where this `NewCreateApplication` came from?

File: ipe.go [hunk 6]
```diff
@@ -75,6 +77,39 @@ func Start(filename string) {
...
+			authHeader := r.Header.Get("Authorization")
+			log.Error(os.Getenv("AUTH_TOKEN"))
+			if authHeader != os.Getenv("AUTH_TOKEN") {
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusUnauthorized)
+				w.Write([]byte(`{ "Error":  "Unauthorized access" }`))
+				return
+			}
+
+			out, err := json.Marshal(inMemoryStorage.GetAllApps())
```
Comment (dimiro1) [inline ipe.go:94]: Don't use the `inMemoryStorage` directly. Use an abstraction on top of `storage.Storage`.

**Comment by dimiro1** [comment]
Hello, @mahavirvataliya I left a few comments.

**Comment by mahavirvataliya** [comment]
Sure @dimiro1  I will look in to it.

**Pull request closed without merging**